### PR TITLE
feat(query): "Encoding Header 'Content-Type' Improperly Defined" for OpenAPI (#3024)

### LINKED ISSUE #3024: Add "Encoding Header 'Content-Type' Improperly Defined" query for OpenAPI
### Platform
OpenAPI

### Description
Encoding Map Key should not define a 'Content-Type' in the 'headers' field. If so, it will be ignored.

### PR DESCRIPTION
**Pull request by rafaela-soares**
Signed-off-by: Rafaela Soares rafaela.soares@checkmarx.com

Closes #3024

**Proposed Changes**
- Added "Encoding Header 'Content-Type' Improperly Defined" query for OpenAPI. It checks if  'encoding[e]' defines 'Content-Type' in the 'headers' field

**Considerations**:
- The Media Type Object exists in Content.
- The Content exists in Responses Object and Request Body Object.
- The Responses Object exists in Component Object and Operation Object.
- The Request Body Object exists in Components Object and Operation Object.

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: assets/queries/openAPI/encoding_header_content_type_improperly_defined/query.rego [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package Cx
+
+import data.generic.openapi as openapi_lib
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) != "undefined"
+
+	header := doc.paths[path][operation].responses[r].content[c].encoding[e].headers[h]
```
Comment (felipe-avelar) [inline assets/queries/openAPI/encoding_header_content_type_improperly_defined/query.rego:9]: I think `.content[c].encoding[e].headers[h]` is duplicated in too many lines... Shouldn't be better create a function `check_content_header` and make checks inside this function?

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Please, check comments

**Review by felipe-avelar — APPROVED** [review]
LGTM

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Pull request merged**